### PR TITLE
[SM-1096] Unify State Variable Naming

### PR DESCRIPTION
**Pull request by coltonhurst**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/SM-1096

## 📔 Objective

The naming around state files and paths is inconsistent. This PR aims to make the usage consistent.

Every mention of state for access token auth should follow this naming logic:

`state_dir` <- This refers to the state directory, where state files are stored. The sdk is currently only aware of state files, directory handling is still being done in bws.
`state_file` <- This refers to a specific state file.

In our code we don't append naming to variables to indicate the type (Hungarian Notation), so I thought it pertinent to truncate `_path` from these, preferring more descriptive `state_dir` and `state_file` for the names.

Thanks @dani-garcia for pointing this out [here](https://github.com/bitwarden/sdk/pull/559#discussion_r1474654682).

⚠️ **Note** ⚠️ This also simplifies the `state_file_dir` key in the `bws` config file to `state_dir`. There will be more changes to this functionality with SM-1174, so I will wait to update the changelog for the next bws version until all this is completed.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by dani-garcia — APPROVED** [review]
LGTM, do we need to update the changelog as well?

File: crates/bws/src/config.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ pub(crate) struct Profile {
     pub server_base: Option<String>,
     pub server_api: Option<String>,
     pub server_identity: Option<String>,
-    pub state_file_dir: Option<String>,
+    pub state_dir: Option<String>,
```
Comment (mzieniukbw) [inline crates/bws/src/config.rs:23]: Would be great to have some test coverage for this.
At the moment we just cover `server_base`

File: languages/go/README.md [hunk 2]
```diff
@@ -36,13 +36,13 @@ bitwardenClient, _ := sdk.NewBitwardenClient(&apiURL, &identityURL)
 
 ### Login
 
-To login using an access token. Define some `statePath` and pass it to use state, or pass `nil`
+To login using an access token. Define some `stateFile` and pass it to use state, or pass `nil`
 instead to not use state.
 
 ```go
-statePath := os.Getenv("STATE_PATH")
+stateFile := os.Getenv("STATE_DIR")
```
Comment (mzieniukbw) [inline languages/go/README.md:43]:
```suggestion
stateFile := os.Getenv("STATE_FILE")
```

**Review by mzieniukbw — CHANGES_REQUESTED** [review]
Found a typo in Go language wrapper README.md.

File: languages/go/README.md (hunk 2 above)
Comment (coltonhurst) [reply languages/go/README.md]: Good find, updated in https://github.com/bitwarden/sdk/commit/b9ea38482bedc79c6ad255ae023571f4455b8248

File: crates/bws/src/config.rs (hunk 1 above)
Comment (coltonhurst) [reply crates/bws/src/config.rs:23]: 100%, we actually already have a ticket for this: https://bitwarden.atlassian.net/browse/SM-1052

I need to go back and work through that ticket.

**Review by dani-garcia — APPROVED** [review]

**Review by mzieniukbw — APPROVED** [review]

**Pull request merged**